Template.php

```diff
@@ -1057,7 +1057,7 @@ public function get_identifiers_from_url($url_sent = NULL) {
           }
           return $this->add_if_new('pmc', $match[1]);
         }
-      } elseif(preg_match("~^https?://citeseerx\.ist\.psu\.edu/viewdoc/(?:summary|download)\?doi=([0-9.]*)(&.+)?~", $url, $match)) {
+      } elseif(preg_match("~^https?://citeseerx\.ist\.psu\.edu/viewdoc/(?:summary|download)(?:\;jsessionid=[^?]+|)\?doi=([0-9.]*)(?:&.+)?~", $url, $match)) {
```

Comment:
Does the `?` need escaping?
```suggestion
      } elseif(preg_match("~^https?://citeseerx\.ist\.psu\.edu/viewdoc/(?:summary|download)(?:\;jsessionid=[^\?]+|)\?doi=([0-9.]*)(?:&.+)?~", $url, $match)) {
```